src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -230,10 +225,37 @@ public static JsonRpc Attach(Stream sendingStream, Stream receivingStream, objec
         }
 
         /// <summary>
+        /// Adds the specified targets as possible objects to invoke when incoming messages are received.
+        /// </summary>
+        /// <param name="targets">Targets to invoke when incoming messages are received.</param>
+        /// <remarks>This method must be called before JsonRpc starts listening for messages.</remarks>
+        public void AddLocalRpcTarget(params object[] targets)
+        {
+            Requires.NotNull(targets, nameof(targets));
+
+            if (this.startedListening)
+            {
+                throw new InvalidOperationException("Cannot attach additional targets once JsonRpc has started listening for messages.");
```

Comment:
Please move the message to `Strings.resx` so it can be localized.
Also, you can reduce these 4 lines to just one by calling `Verify.Operation`